utilities/general-helper-functions.metta

```diff
@@ -46,4 +46,90 @@
         )
 )
 )
+)
+
+(= (getchildren $expr $literals $nonLiterals)
+    (if (== $expr ())
+        ($literals $nonLiterals)
+        (let* 
+            (   
+                ($head (car-atom $expr))
+                ($tail (cdr-atom $expr))
+                ($isLiteral (or (== (get-metatype $head) Symbol) (unify $head (NOT $_) True False)))
+            )
+            (if (or (== $head AND) (== $head OR)) 
+                (getchildren $tail $literals $nonLiterals)
+
+                (if $isLiteral 
+                    (getchildren $tail (concatTuple $literals ($head)) $nonLiterals)
+                    (getchildren $tail $literals (concatTuple $nonLiterals ($head)))        
+                )
+            )
+        )
+    )
+)
+
+(= (getGsetAndChildren $expr) (getchildren $expr () ()))
+
+(= (findIntersection $tuple1 $tuple2)
```

Comment:
This function seems to be useful for any type of atom right? So why not rename it to `atomIntersection`. This makes the function name descriptive and anyone knows what this function behavior could be by only looking at it's name.